packages/vite/src/node/plugins/assetImportMetaUrl.ts

```diff
@@ -74,8 +68,9 @@ export function assetImportMetaUrlPlugin(config: ResolvedConfig): Plugin {
           // Get final asset URL. Catch error if the file does not exist,
           // in which we can resort to the initial URL and let it resolve in runtime
           const builtUrl = await fileToUrl(file, config, this).catch(() => {
+            const truthExp = code.slice(index, index + exp.length)
             config.logger.warnOnce(
-              `\n${exp} doesn't exist at build time, it will remain unchanged to be resolved at runtime`
+              `\n${truthExp} doesn't exist at build time, it will remain unchanged to be resolved at runtime`
```

Comment:
I'm curious what this change fixes. Would appreciate if you can shed some light on this.

Reply:
Because the string here has been replaced, the string with words must be taken from the original string. 

`emptyString` will let the string to 

```ts
"aaaa"
```

<hr>

```ts
"\0\0\0\0"
```

Reply:
Ah I see, makes sense then. A tiny nitpick, but perhaps we can rename `truthExp` to something like `rawExp`. I find `truthExp` a bit confusing 😅

Reply:
ok, it's right